### PR TITLE
Restyled launcher screen

### PR DESCRIPTION
**Pull request by BoatsAreRockable**
The launcher now matches the dark theme (issue #57)

launcherdemo3

@le717 asked to be mentioned so hi! I'd draw your attention to how the version number is obtained from package.json, I have a feeling that you will have something to say about it.

### CODE REVIEW

**Comment by le717** [comment]
Just quickly browsing: I like the design (rioforce can comment if he wishes). Code is OK though there are some points I wish to call out (not at the moment) for review.

> I have a feeling that you will have something to say about it.

You learn fast. :stuck_out_tongue: That said, I say stuff because I really want Boats to succeed and have well-written code to support lots of features and usage for a long time to come and not incur technical debt, and to do that you often have to be "picky" about how things are implemented. :)

**Comment by rioforce** [comment]
So far, it looks good. I'd say go ahead and merge it. I think I might tweak the scale of the buttons though.

**Comment by le717** [comment]
Sorry for the delay. I still want to code review this. I'll get to it ASAP. :)

**Comment by le717** [comment]
Alright! College is over, finally have time to code review this! Sorry for the wait. Here we go!

File: app/css/animator.css [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 html {
   height: 100%;
-  min-width: 590px;
-  min-height: 590px;
```
Comment (le717) [inline app/css/animator.css]: What was the purpose behind removing this? They prevent the layout from resizing beyond the minimum size utterly breaking.

File: app/css/animator.css [hunk 2]
```diff
@@ -101,13 +99,13 @@ a {
...
-  padding-right: 1em;
-  height: 100%;
-  width: 19em;
+    position: relative;
+    float: right;
+    color: #d3d3d3;
+    z-index: 2;
+    padding-right: 1em;
+    height: 100%;
+    width: 19em;
```
Comment (le717) [inline app/css/animator.css]: *\sigh*  I really need to propose some code style standards.  We should have had those already. In the meantime, please revert this section back to 2 space (trying to keep out unnecessary whitespace changes like this one).

File: app/css/animator.css (hunk 1 above)
Comment (BoatsAreRockable) [inline app/css/animator.css]: I added min-width and min-height properties to package.json to replace this.

File: app/css/animator.css [hunk 3]
```diff
@@ -444,9 +440,7 @@ a {
   -webkit-appearance: none;
   margin-top: -9.5px;
 }
-#input-onion-skin-opacity:focus::-webkit-slider-runnable-track {
-  background-color: #D9D9D9;
-}
+#input-onion-skin-opacity:focus::-webkit-slider-runnable-track { background-color: #D9D9D9; }
```
Comment (le717) [inline app/css/animator.css]: Just because this selector is so long, I would be fine with keeping it in the expanded format.

File: app/js/main.js [hunk 4]
```diff
@@ -659,7 +666,7 @@ function saveFrame(id) {
 
     // 1K+ frames have been captured
     if (id >= 1000) {
-      fileName = 'frame_${id}';
+      fileName = `frame_${id}`;
```
Comment (le717) [inline app/js/main.js]: Niiiice. :stuck_out_tongue:

File: app/index.html [hunk 5]
```diff
@@ -4,28 +4,51 @@
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
     <title>Boats Animator</title>
-    <link rel="stylesheet" href="css/index.css">
+    <link rel="stylesheet" href="css/font-awesome.min.css">
+    <link rel="stylesheet" href="css/fonts.css">
+    <link rel="stylesheet" href="css/animator.css">
```
Comment (le717) [inline app/index.html]: I believe rioforce (or was that me? I forget) created `index.css` because there was conflicting CSS between the two windows and it makes more sense to load just the CSS needed for that window and not everything else. By removing the reference to `index.css` and stuffing everything in `animator.css`, the deliberate separation has been reverted. Please change it back to `index.css` and move the splash-specific CSS style into it.

File: app/css/animator.css [hunk 6]
```diff
@@ -505,4 +499,91 @@ a {
 
 ::-webkit-scrollbar-corner       { background-color: #000; }
 
-::-webkit-resizer                { }
\ No newline at end of file
+::-webkit-resizer                { }
+
+/* ======== Preference Windows ======== */
```
Comment (le717) [inline app/css/animator.css]: Please move all this (Preference Windows???, Projects Area, and Recent News to `index.css`. You will also need to copy the scrollbar styling into it.

File: app/index.html [hunk 7]
```diff
@@ -4,28 +4,51 @@
...
+            </div>
+        </div>
+    </div>
+    <div id="main-event">
+        <div class="content">
+            <h2>Projects</h2>
+            <ul id="projects-container">
+                <li><i class="fa fa-folder-open-o"></i> <a href="#">Recent Project 1</a></li>
+                <li><i class="fa fa-folder-open-o"></i> <a href="#">Recent Project 2</a></li>
+                <li><i class="fa fa-folder-open-o"></i> <a href="#">Recent Project 3</a></li>
```
Comment (le717) [inline app/index.html]: Be aware this HTML will be dynamically injected instead of hard-coded like this. You'll just want to comment out (instead of `display: none`) the projects stuff right before merging.

File: app/index.html [hunk 8]
```diff
@@ -4,28 +4,51 @@
...
+            </ul>
+            <button id="new-project" onclick="openAnimator()"><i class="fa fa-file-o"></i><span>New Project</span></button>
+            <button id="load-project" onclick="alert('This feature is coming soon!')"><i class="fa fa-folder-o"></i><span>Load Project</span></button>
+        </div>
+    </div>
+    
+    <div id="statusBar">
+        <ul>
+            <li>Version <span id="version"></span></li>
```
Comment (le717) [inline app/index.html]: Change the ID to `app-version`.

File: app/js/main.js [hunk 9]
```diff
@@ -84,7 +84,10 @@ var width  = 640,
     confirmContainer    = document.querySelector("#confirm-container"),
     confirmText         = document.querySelector("#confirm-text"),
     btnConfirmOK        = document.querySelector("#confirm-container #btn-OK"),
-    btnConfirmCancel    = document.querySelector("#confirm-container #btn-cancel");
+    btnConfirmCancel    = document.querySelector("#confirm-container #btn-cancel"),
+
+    // Launcher window
+    launcherVersion     = document.querySelector("#version");
```
Comment (le717) [inline app/js/main.js]: Change selector to `#app-version`.

File: app/css/animator.css (hunk 1 above)
Comment (le717) [inline app/css/animator.css]: I see that now. Thank you. :)

File: app/js/main.js [hunk 10]
```diff
@@ -955,3 +963,12 @@ function dev() {
 function reload() {
     win.reloadDev();
 }
+
+/**
+ * Get version number from package.json
+ */
+fs.readFile("package.json", "utf8", function (err, data) {
```
Comment (le717) [inline app/js/main.js]: If we are going to parse `package.json`, we are going to get more information from it then just the version. However, I think this is fine for now. :)

File: app/index.html [hunk 11]
```diff
@@ -4,28 +4,51 @@
...
+    <link rel="stylesheet" href="css/font-awesome.min.css">
+    <link rel="stylesheet" href="css/fonts.css">
+    <link rel="stylesheet" href="css/animator.css">
 </head>
 
-<body class="index" onload="canDisplayNews()">
-    <header class="app-info">
-        <h1>Boats Animator</h1>
-        <footer>Copyright (C) Charlie Lee 2015 <span style="float: right; text-align: right; padding: 0 4%;"><a href="#" onclick="gui.Shell.openExternal('http://charlielee.uk')">charlielee.uk</a></span></footer>
+<body class="animator" onload="canDisplayNews()">
```
Comment (le717) [inline app/index.html]: Again, this class on the body was to separate the two stylesheets. However, neither `.animator` or `.index` is used, so you can just remove this class attribute.

**Comment by le717** [comment]
Alright, I think I am done with the review. :) Let me know when you push the changes. I'll take one last look and it should be ready to merge.

File: app/css/animator.css (hunk 2 above)
Comment (BoatsAreRockable) [inline app/css/animator.css]: Yes, this quite a mess right now! Will revert.

File: app/css/animator.css (hunk 6 above)
Comment (BoatsAreRockable) [inline app/css/animator.css]: Will do. There are a few common common styles such as `#sidebar` I've taken from animator.css, I assume it would be better to continue pulling these in from animator.css?

File: app/index.html (hunk 5 above)
Comment (BoatsAreRockable) [inline app/index.html]: Okay.

File: app/index.html (hunk 7 above)
Comment (BoatsAreRockable) [inline app/index.html]: Ah yes this purely to give a visual demo of things. Will comment out.

**Comment by BoatsAreRockable** [comment]
Thank you for very much for doing this, it is most appreciated. I shall aim to implement these changes tomorrow. I still have a couple of weeks of school left but hopefully after that I can work on a few things.

File: app/css/animator.css (hunk 6 above)
Comment (le717) [inline app/css/animator.css]: Eh, the sidebar has been drastically reduced and will continue to be cut... I think it'll be fine to be copied over into index.css.

**Comment by BoatsAreRockable** [comment]
@le717 I've finished pushing the changes.

**Comment by le717** [comment]
Other my one comment everything looks good. Let me know when you have that sorted out.

**Comment by BoatsAreRockable** [comment]
@le717 sorted it out, animator.css is no longer loaded into index.html.

**Comment by le717** [comment]
:+1: :shipit: :P 

(Read: Thumbs up. Ship it. :P)
